packages/visx-annotation/src/components/Label.tsx

```diff
@@ -67,7 +69,78 @@ function getCompletePadding(padding: LabelProps['backgroundPadding']) {
   return { ...DEFAULT_PADDING, ...padding };
 }
 
-export default function Label({
+export default function Label({ ...props }: LabelProps) {
+  if (props.children) {
+    return <ForeignObjectLabel {...props} />;
+  }
+  return <SvgLabel {...props} />;
+}
+
+function ForeignObjectLabel({
+  anchorLineStroke = '#222',
+  children,
+  className,
+  horizontalAnchor: propsHorizontalAnchor,
+  resizeObserverPolyfill,
+  showAnchorLine = true,
+  verticalAnchor: propsVerticalAnchor,
+  x: propsX,
+  y: propsY,
+}: LabelProps) {
+  // we must measure the rendered title + subtitle to compute container height
+  const [labelRef, titleBounds] = useMeasure({
+    polyfill: resizeObserverPolyfill,
+  });
+  const { width, height } = titleBounds;
+
+  // if props are provided, they take precedence over context
+  const { x = 0, y = 0, dx = 0, dy = 0 } = useContext(AnnotationContext);
+
+  // offset container position based on horizontal + vertical anchor
+  const horizontalAnchor =
+    propsHorizontalAnchor || (Math.abs(dx) < Math.abs(dy) ? 'middle' : dx > 0 ? 'start' : 'end');
+  const verticalAnchor =
+    propsVerticalAnchor || (Math.abs(dx) > Math.abs(dy) ? 'middle' : dy > 0 ? 'start' : 'end');
+
+  const containerCoords = useMemo(() => {
+    let adjustedX: number = propsX == null ? x + dx : propsX;
+    let adjustedY: number = propsY == null ? y + dy : propsY;
+
+    if (horizontalAnchor === 'middle') adjustedX -= width / 2;
+    if (horizontalAnchor === 'end') adjustedX -= width;
+    if (verticalAnchor === 'middle') adjustedY -= height / 2;
+    if (verticalAnchor === 'end') adjustedY -= height;
+
+    return { x: adjustedX, y: adjustedY };
+  }, [propsX, x, dx, propsY, y, dy, horizontalAnchor, verticalAnchor, width, height]);
+
+  return (
+    <Group
+      top={containerCoords.y}
+      left={containerCoords.x}
+      pointerEvents="none"
+      className={cx('visx-annotationlabel', className)}
+    >
+      {showAnchorLine && (
+        <AnchorLine
+          anchorLineOrientation={Math.abs(dx) > Math.abs(dy) ? 'vertical' : 'horizontal'}
+          anchorLineStroke={anchorLineStroke}
+          verticalAnchor={verticalAnchor}
+          horizontalAnchor={horizontalAnchor}
+          width={width}
+          height={height}
+        />
+      )}
+      <foreignObject width={width} height={height} overflow="visible">
+        <div ref={labelRef} style={{ display: 'inline-block' }}>
```

Comment:
can we pull the `{ display: 'inline-block' }` style into a `const` outside the function since it doesn't change?